File: gateway/enforcer/org.wso2.apk.enforcer.commons/src/main/java/org/wso2/apk/enforcer/commons/model/AIProvider.java
package org.wso2.apk.enforcer.commons.model;

// AIProvider is used to provide the AI model to the enforcer
public class AIProvider {
    private String providerName;
    private String providerAPIVersion;
    private String organization;

    private Boolean enabled = false;

    private ValueDetails model;

    private ValueDetails promptTokens;

    private ValueDetails completionToken;

    private ValueDetails totalToken;

    // Get provider name
    public String getProviderName() {
        return providerName;
    }

    // Get provider API version
    public String getProviderAPIVersion() {
        return providerAPIVersion;
    }

    // Get enabled
    public Boolean getEnabled() {
        return enabled;
    }

    // Get organization
    public String getOrganization() {
        return organization;
    }

    // Get model
    public ValueDetails getModel() {
        return model;
    }

    // Get prompt tokens
    public ValueDetails getPromptTokens() {
        return promptTokens;
    }

    // Get completion token
    public ValueDetails getCompletionToken() {
        return completionToken;
    }

    // Get total token
    public ValueDetails getTotalToken() {
        return totalToken;
    }

    // Set provider name
    public void setProviderName(String providerName) {
        this.providerName = providerName;
    }

    // Set provider API version
    public void setProviderAPIVersion(String providerAPIVersion) {
        this.providerAPIVersion = providerAPIVersion;
    }

    // Set enabled
    public void setEnabled(Boolean enabled) {
        this.enabled = enabled;
    }

    // Set organization
    public void setOrganization(String organization) {
        this.organization = organization;
    }

    // Set model
    public void setModel(ValueDetails model) {
        this.model = model;
    }

    // Set prompt tokens
    public void setPromptTokens(ValueDetails promptTokens) {
        this.promptTokens = promptTokens;
    }

    // Set completion token
    public void setCompletionToken(ValueDetails completionToken) {
        this.completionToken = completionToken;
    }

    // Set total token
    public void setTotalToken(ValueDetails totalToken) {
        this.totalToken = totalToken;
    }

}

File: gateway/enforcer/internal/dto/ai_provider.go
/*
 *  Copyright (c) 2025, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

package dto

// AIProvider represents the AI provider details.
type AIProvider struct {
	ProviderName       string        `json:"providerName"`       // Name of the AI provider
	ProviderAPIVersion string        `json:"providerAPIVersion"` // API version of the AI provider
	Organization       string        `json:"organization"`       // Organization associated with the provider
	Enabled            bool          `json:"enabled"`            // Whether the provider is enabled
	SupportedModels    []string      `json:"supportedModels"`    // Supported models
	Model              *ValueDetails `json:"model"`              // Model details
	PromptTokens       *ValueDetails `json:"promptTokens"`       // Prompt token details
	CompletionToken    *ValueDetails `json:"completionToken"`    // Completion token details
	TotalToken         *ValueDetails `json:"totalToken"`         // Total token details
}
